packages/jsActions/nanoflow-actions-native/src/geolocation/GetCurrentLocation.ts

```diff
@@ -52,14 +68,20 @@ export async function GetCurrentLocation(
             });
         }
 
-        function onError(error: GeolocationError): void {
+        function onError(error: GeolocationError | GeoError): void {
             return reject(new Error(error.message));
         }
 
-        function getOptions(): GeolocationOptions {
-            const timeoutNumber = timeout && Number(timeout.toString());
+        function getOptions(): GeolocationOptions | GeoOptions {
+            let timeoutNumber = timeout && Number(timeout.toString());
             const maximumAgeNumber = maximumAge && Number(maximumAge.toString());
 
+            if (timeoutNumber === undefined && Platform.OS === "ios") {
```

Comment:
It's helpful for other developers if you describe why this is handled this way for IOS with a comment

Reply:
I added a few comments.